tendermint/src/account.rs

```diff
@@ -75,12 +80,10 @@ impl From<secp256k1::PublicKey> for Id {
 }
 
 // SHA256(pk)[:20]
-impl From<ed25519::PublicKey> for Id {
-    fn from(pk: ed25519::PublicKey) -> Id {
+impl From<Ed25519PublicKey> for Id {
+    fn from(pk: Ed25519PublicKey) -> Id {
         let digest = Sha256::digest(pk.as_bytes());
-        let mut bytes = [0u8; LENGTH];
-        bytes.copy_from_slice(&digest[..LENGTH]);
-        Id(bytes)
+        Id(digest[..LENGTH].try_into().unwrap())
```

Comment:
If this is fallible now, should we try to rework it into a `TryFrom`?

Reply:
The implementation is infallible. It's doing a slice to byte array conversion, and the potential failure case would be the length of the slice doesn't match the length of the destination array. Since both the slice and the array are sized using the `LENGTH` constant, I don't foresee this being a problem.

Note that `copy_from_slice` will panic in this case as well, it's just not as visible.